### DISCUSSION TITLE
Need recommendations for HW specs for running Airsim

### DISCUSSION THREAD

**Discussion by ksat4dev** [microsoft/AirSim#3283] [Support Q&A] [answered]
Hello,

I am trying to run a SITL using PX4-Airsim on a Linux machine. After having read the documentation, it seems Airsim is more realistic both visually and physically. So, I am just wondering what could be the recommended specifications? PS: This is for serious use and not a hobby.

There was no explicit mention of the specification other than a beefy GPU. 

Is Desktop better choice for Airsim than Laptop?
Is NVIDIA GeForce GTX 1050 Ti good? or RTX 3060 Ti preferred?
Is 32 GB Ram on an x64 type computer good enough? 

I would appreciate any device suggestion that might have the right combination among the specs asked above.

Thanks in advance.

**Comment by ahmed-elsaharti** [comment]
Hi @ksat4dev ,
AFAIK (and as hinted [here](https://github.com/microsoft/AirSim/blob/master/docs/use_precompiled.md#dont-have-good-gpu)) the GPU requirement mainly depends on your UE environment. In addition to the tips mentioned in [docs](https://github.com/microsoft/AirSim/blob/master/docs/use_precompiled.md#dont-have-good-gpu), I've seen people run AirSim in headless mode to avoid rendering the scene visually.

In general, I've had multiple colleagues run a couple of moderately demanding environments on GeForce 10 series GPUs (mobile in their cases). But if RTX 3060 is an option I'd go for it without blinking an eye. Here's a comparison between the GPUs you mentioned https://gpu.userbenchmark.com/Compare/Nvidia-RTX-3060-Ti-vs-Nvidia-GTX-1050-Ti/4090vs3649

Also, I would look into getting a pixhawk and running in HITL mode since SITL seems to be having issues (#3260 #2647) at the moment.

**Reply by ksat4dev** [reply]
Hi ahmed,

                  Thanks for your response and also for the heads up regarding issues with SITL. I am facing the same as the one in #2647. I will probably try switching over to HITL. Appreciate it!

**Comment by Zartris** [comment, marked as answer]
Hi @ksat4dev 
I have been using Airsim for a while now, and I think I can help you with some of your questions:
**Is Desktop better choice for Airsim than Laptop?**
- If you are not using this as a hobby I would always go for the desktop. It is better for cooling and when you start to use multiple programs like ROS and PX4 it is starting to be very hot.

**Is NVIDIA GeForce GTX 1050 Ti good? Or RTX 3060 Ti preferred?**
- We have a 1050 Ti, but we are now looking into upgrading to 3060 Ti since we start seen limitation when working with bigger environments using complex lighting. I'm not sure of the extent of the projects you have, but it's better to do it right the first time.

**Is 32 GB Ram on a x64 type computer good enough?**
- 32 should be enough, but again that depends on the project you have. If you are working in a very large environment, RAM is very much needed so go for 64 instead. It does not affect the running of AirSim that much, but more ram will help you develop your own environments in Unreal Engine and Unity. (Unreal uses a lot of ram, where Unity is much more friendly).

** NOTE ** 
You said that you have read docs, but just to be sure [hardrive and ram](https://microsoft.github.io/AirSim/hard_drive/).
Just like they describe here an important factor is also a fast SSD for when your RAM is not enough. If you are running multiple drones at the same time and working with images, this will be needed.

**Reply by ksat4dev** [reply]
Hi Zartris,

                Thank you so much, this is really helpful!
